static_src/test/unit/util/cf_api.spec.js

```diff
@@ -470,6 +470,29 @@ describe('cfApi', function() {
     });
   });
 
+  describe('fetchAppAll()', function () {
+    it('should call fetchApp() and fetchAppStats()', function () {
+      const guid = 'shouldCallTwoFnsGuid';
+      const fetchAppSpy = sandbox.stub(cfApi, 'fetchApp');
+      const fetchAppStatsSpy = sandbox.stub(cfApi, 'fetchAppStats');
+
+      cfApi.fetchAppAll(guid);
+
+      expect(fetchAppSpy).toHaveBeenCalledOnce();
+      expect(fetchAppStatsSpy).toHaveBeenCalledOnce();
+    });
+
+    it('should call the receivedAppAll app action', function (done) {
+      const guid = 'shouldCallActionCreatorGuid';
+      const actionCreatorSpy = sandbox.spy(appActions, 'receivedAppAll');
+
+      cfApi.fetchAppAll(guid).then(() => {
+        expect(actionCreatorSpy).toHaveBeenCalled();
```

Comment:
@msecret this test would always fail if I was expecting the spy `toHaveBeenCalledOnce` with a message that it had been called twice, but I can't figure out why.

From looking at the way this test executes and the code in `cfApi.fetchAppAll()` can you see what makes the `appActions.receivedAppAll` get called twice?